packages/auth0_api_python/src/auth0_api_python/api_client.py

```diff
@@ -14,46 +29,210 @@ class ApiClient:
     """
 
     def __init__(self, options: ApiClientOptions):
-
         if not options.domain:
             raise MissingRequiredArgumentError("domain")
         if not options.audience:
             raise MissingRequiredArgumentError("audience")
 
         self.options = options
-        self._metadata: Optional[Dict[str, Any]] = None
-        self._jwks_data: Optional[Dict[str, Any]] = None
+        self._metadata: Optional[dict[str, Any]] = None
+        self._jwks_data: Optional[dict[str, Any]] = None
 
         self._jwt = JsonWebToken(["RS256"])
 
-    async def _discover(self) -> Dict[str, Any]:
-        """Lazy-load OIDC discovery metadata."""
-        if self._metadata is None:
-            self._metadata = await fetch_oidc_metadata(
-                domain=self.options.domain,
-                custom_fetch=self.options.custom_fetch
-            )
-        return self._metadata
+        self._dpop_algorithms = ["ES256"]
+        self._dpop_jwt = JsonWebToken(self._dpop_algorithms)
 
-    async def _load_jwks(self) -> Dict[str, Any]:
-        """Fetches and caches JWKS data from the OIDC metadata."""
-        if self._jwks_data is None:
-            metadata = await self._discover()
-            jwks_uri = metadata["jwks_uri"]
-            self._jwks_data = await fetch_jwks(
-                jwks_uri=jwks_uri, 
-                custom_fetch=self.options.custom_fetch
+
+    async def verify_request(
+        self,
+        headers: dict[str, str],
+        http_method: Optional[str] = None,
+        http_url: Optional[str] = None
+    ) -> dict[str, Any]:
+        """
+        Dispatch based on Authorization scheme:
+          • If scheme is 'DPoP', verifies both access token and DPoP proof
+          • If scheme is 'Bearer', verifies only the access token
+
+        Args:
+            headers: HTTP headers dict containing:
+                - "authorization": The Authorization header value (required)
+                - "dpop": The DPoP proof header value (required for DPoP)
+            http_method: The HTTP method (required for DPoP)
+            http_url: The HTTP URL (required for DPoP)
+
+        Returns:
+            The decoded access token claims
+
+        Raises:
+            MissingRequiredArgumentError: If required args are missing
+            InvalidAuthSchemeError: If an unsupported scheme is provided
+            InvalidDpopProofError: If DPoP verification fails
+            VerifyAccessTokenError: If access token verification fails
+        """
+        authorization_header = headers.get("authorization", "")
+        dpop_proof = headers.get("dpop")
```

Comment:
For the `Authorization` header, there's what seems to be a fallback value (an empty string).
Is it necessary?
Especially since for the DPoP header it isn't used.